src/integrations/watchers/indexable-ancestor-watcher.php

```diff
@@ -195,26 +184,29 @@ protected function update_hierarchy_and_permalink( $indexable ) {
 	/**
 	 * Retrieves the object id's for a term based on the term-post relationship.
 	 *
-	 * @param int         $term_id          The term to get the object id's for.
-	 * @param Indexable[] $child_indexables The child indexables.
+	 * @param int              $term_id          The term to get the object id's for.
+	 * @param array<Indexable> $child_indexables The child indexables.
 	 *
-	 * @return array List with object ids for the term.
+	 * @return array<int> List with object ids for the term.
 	 */
 	protected function get_object_ids_for_term( $term_id, $child_indexables ) {
+		global $wpdb;
+
 		$filter_terms = static function ( $child ) {
 			return $child->object_type === 'term';
 		};
 
 		$child_terms      = \array_filter( $child_indexables, $filter_terms );
-		$child_object_ids = \wp_list_pluck( $child_terms, 'object_id' );
+		$child_object_ids = \array_merge( [ $term_id ], \wp_list_pluck( $child_terms, 'object_id' ) );
 
 		// Get the term-taxonomy id's for the term and its children.
-		$term_taxonomy_ids = $this->wpdb->get_col(
-			$this->wpdb->prepare(
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
+		$term_taxonomy_ids = $wpdb->get_col(
+			$wpdb->prepare(
 				'SELECT term_taxonomy_id
```

Comment:
Let's use `%i` placeholders for attribute names, too (e.g. `term_taxonomy_id` 🙂

Reply:
No need to add placeholder to hardcoded attributes names.